internal/system/impl/sqlstore.go

```diff
@@ -68,7 +70,15 @@ func (s *SystemSQLStoreService) GetTableMetadata(
 	}
 	table, err := store.GetTable(ctx, id)
 	if err != nil {
-		return sqlstore.TableMetadata{}, fmt.Errorf("error fetching the table: %s", err)
+		if !errors.Is(err, sql.ErrNoRows) {
+			return sqlstore.TableMetadata{}, fmt.Errorf("error fetching the table: %s", err)
```

Comment:
If a database error occurs, we're not returning any table metadata and still returning an error. Kind of strange to return NFT metadata for a database error.